repos/system_upgrade/el7toel8/actors/checkkdegnome/actor.py

```diff
@@ -0,0 +1,167 @@
+                    reporting.Summary("With only KDE installed, there would be no other desktop  after upgrade."),
+                    reporting.Severity(reporting.Severity.HIGH),
+                    reporting.Tags([
+                        reporting.Tags.UPGRADE_PROCESS
+                    ]),
+                    reporting.Flags([
+                        reporting.Flags.INHIBITOR
+                    ]),
+                    reporting.Remediation(
+                        hint="Install GNOME desktop to be able to upgrade.")
+                    ])
+                return
+
+            # Assume GNOME is installed in this state
+
+            user = os.environ.get("USER")
+            default_xsession = self.get_xsession("/var/lib/AccountsService/users/{0}".format(user))
+            if not default_xsession:
+                self.log.warning("Unable to detect default session.")
+            else:
+                if "plasma" in default_xsession:  # using in because there can be some white spaces
+                    self.log.info("KDE used as default session.")
+                    self.log.info("Upgrade can be performed, but KDE desktop will be removed in favor of GNOME")
+                    create_report([
+                        reporting.Title("Upgrade can be performed, but KDE will be uninstalled."),
+                        reporting.Summary("KDE has to be uninstalled in favor of GNOME to perform the upgrade."),
+                        reporting.Severity(reporting.Severity.MEDIUM),
+                        reporting.Tags([
+                            reporting.Tags.UPGRADE_PROCESS
+                        ]),
+                        reporting.Remediation(
+                            hint="KDE has to be removed, no other solution is possible.")
+                        ])
+                    return  # upgrade can be performed, but user will loose KDE desktop in favor of GNOME desktop
+                else:
+                    self.log.info("GNOME used as default session. Continuing with the upgrade.")
+            self.log.info("----------------------------------")
+
+        # At this state we can assume that KDE desktop as such is not installed or used and we just need to
+        # detect whether any KDE/Qt app is actively used to inform user that the application will be removed
+        # during the upgrade process
+
+        base_kde_apps = ("kde-baseapps",
+                         "okular",
+                         "ark",
+                         "kdepim",
+                         "konsole",
+                         "gwenview",
+                         "kdenetwork",
+                         "kate", "kwrite")
+        self.log.info("  Detecting installed KDE apps  ")
+        self.log.info("================================")
+
+        for app in base_kde_apps:
+            if self.is_installed(app) == _ret_OK:  # return code of "rpm -q app" command
```

Comment:
Having '0' inplace instead of _ret_OK makes more sense imho. Also I'd rather move the whole logic of check into is_installed method and have it return True/False and not an exit code.